### PR TITLE
Add `ginibre` metric to `random_density_matrix` and `bcsz` measure to `random_quantum_channel`

### PR DESCRIPTION
**Pull request by renatomello**
Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

### CODE REVIEW

**Review by AlejandroSopena — APPROVED** [review]
Thanks @renatomello. Just out of curiosity: I've noticed that you've implemented the first method proposed in [Random quantum operations](https://doi.org/10.1016/j.physleta.2008.11.043). Why that method instead of the second one? Anyway, I agree with your implementation.

**Comment by renatomello** [comment]
> Thanks @renatomello. Just out of curiosity: I've noticed that you've implemented the first method proposed in [Random quantum operations](https://doi.org/10.1016/j.physleta.2008.11.043) for the `bcsz` measure. Why that method instead of the second one? Anyway, I agree with your implementation.

No particular reason, to be honest.

**Review by stavros11 — APPROVED** [review]

**Comment by AleCandido** [comment]
This PR has two approvals, but it would be nice to implement @stavros11 comments before merging (they are pretty simple anyhow).

I'm just writing it to avoid a quick merge during a meeting. In case, could you please just remind us about this? @renatomello

**Comment by renatomello** [comment]
> This PR has two approvals, but it would be nice to implement @stavros11 comments before merging (they are pretty simple anyhow).
> 
> I'm just writing it to avoid a quick merge during a meeting. In case, could you please just remind us about this? @renatomello

I will make the changes soon, but I'm just caught up on something else right now

**Comment by AleCandido** [comment]
> I will make the changes soon, but I'm just caught up on something else right now

No worries, this is not blocking anything, and there is no further reason to be in a hurry :)

**Comment by scarrazza** [comment]
@renatomello, I am planning to release 0.2.1 soon (say tomorrow), could be great if you managed to make it.

File: src/qibo/quantum_info/random_ensembles.py
Comment (renatomello) [reply src/qibo/quantum_info/random_ensembles.py:302]: AFAIK the garbage collector does clean all variables that a not the returned one, including `numpy` arrays. Moved that part of the function to a helper function, as suggested.

**Pull request merged**
